src/components/card/_macro.njk

```diff
@@ -24,15 +24,15 @@
 
     {%- else -%}
 
-      <h{{ titleSize }} class="ons-card__title {{ params.titleClasses | default('ons-u-fs-m') }}" id="{{ params.id }}">
-        {%- if params.url -%}
-          <a class="ons-card__link" href="{{ params.url }}">
-        {%- endif -%}
+      {%- if params.url -%}
+        <a class="ons-card__link" href="{{ params.url }}">
+      {%- endif -%}
+        <h{{ titleSize }} class="ons-card__title ons-card__title--reorder {{ params.titleClasses | default('ons-u-fs-m')}} ons-card__remove--margin" id="{{ params.id }}">
           {{ params.title }}
-        {%- if params.url -%}
-          </a>
-        {%- endif -%}
-      </h{{ titleSize }}>
+        </h{{ titleSize }}>
+      {%- if params.url -%}
+        </a>
+      {%- endif -%}
```

Comment:
Now you've made this change to reorder the `<a>` and `<h#>` there is a lot more repeated code here so you could simplify this and move the `{%- if params.image -%}` and put it above the `{% if params.image.smallSrc %}` line. Then you can remove a lot of the code around it.